### PR TITLE
disabling reduction fusion on size-0 tensors

### LINKED ISSUE #1466: Scheduler hits floating point exception (dividing by 0)
### 🐛 Describe the bug

I'm hitting issue when we have size-0 rank 1 tensors going through a reduction kernel.

```
import torch                                                                                                            
                                                                                                                        
@torch.jit.script                                                                                                       
def f(t):                                                                                                               
    return torch.nn.functional.softmax(t, dim=-1)                                                                       
                                                                                                                        
t0 = torch.randn(0, device="cuda")                                                                                      
s0 = torch.tensor(1.0, device="cuda")                                                                                   
                                                                                                                                                                                                                   
torch._C._jit_set_nvfuser_single_node_mode(True)                                                                        
                                                                                                                        
with torch.jit.fuser("fuser2"):                                                                                         
    f(t0)                                                                                                               
    f(t0)                                                                                                               
    f(t0)                                                                                                               
    print(f.graph_for(t0)) 
```

I tried to patch it with gdb but that just exposed more divide by 0 :fearful: 

### Versions

Reproed on top of devel

```
commit 592fd4762a8004c71d4acd5d885e04a85ccceae0 (csarofeen/devel)
Author: jjsjann123 <jiej@nvidia.com>
Date:   Fri Feb 11 23:52:03 2022 -0800

    disabling 0-dim cuda tensor reduction/normalization (#1453)
    
    1. Disables normalizatoin/reduction for 0-dim cuda inputs.
    2. Patches fusion logic on reduction parsing rule
    
    We are doing this because normalization for 0-dim behavior is tricky to justify (?!) and seems implementation specific. So we just leave this for eager.

```

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes #1466 

Note that this is a WAR, not a proper fix on codegen to handle size-0

### CODE REVIEW

**Comment by csarofeen** [comment]
Asserts for this by the way: https://github.com/csarofeen/pytorch/pull/1470

**Review by csarofeen — APPROVED** [review]
LGTM
